source/fab/steps/psyclone.py

```diff
@@ -79,7 +80,24 @@ class Psyclone(Step):
     def __init__(self, name=None, kernel_roots=None,
                  transformation_script: Optional[Path] = None,
                  cli_args: Optional[List[str]] = None,
-                 source_getter: Optional[ArtefactsGetter] = None):
+                 source_getter: Optional[ArtefactsGetter] = None,
+                 overrides_folder: Optional[Path] = None):
+        """
+        :param name:
+            Human friendly name for logger output.
+        :param kernel_root:
+            Folder containing kernel files. Must be part of the analysed source code.
+        :param transformation_script:
+            The Python transformation script.
+        :param cli_args:
+            Passed through to the psyclone cli tool.
+        :param source_getter:
+            Optional override for getting input files from the artefact store.
+        :param overrides_folder:
+            Optional folder containing hand-crafted override files.
+            Must be part of the analysed source code.
```

Comment:
Is there a potential problem with having two files which define the same symbols? Can you think of any scenario when there might be two copies of the symbol floating around.

If the source has to have been analysed before override happens doesn't that mean there's a copy of the symbol in the override file and one in the generated (and then overridden) output?

Reply:
> If the source has to have been analysed before override happens

It's the other way round, we have to run psyclone before we run the analysis step because the psyclone step produces new Fortran. We'll make this comment clearer.